File: cfg/KuroApi.py
import requests
import time


class KuroApi:
    BASE_URLS = {
        "game_roles": "https://api.kurobbs.com/gamer/role/default",
        "init_signin": "https://api.kurobbs.com/encourage/signIn/initSignInV2",
        "game_signin": "https://api.kurobbs.com/encourage/signIn/v2",
        "sign_record": "https://api.kurobbs.com/encourage/signIn/queryRecordV2",
        "mc_widget": "https://api.kurobbs.com/gamer/widget/game3/getData",
        "zs_widget": "https://api.kurobbs.com/gamer/widget/game2/getData",
        'zs_month': "https://api.kurobbs.com/gamer/resource/month",
        "forum_signin": "https://api.kurobbs.com/user/signIn",
        "post_list": "https://api.kurobbs.com/forum/list",
        "post_detail": "https://api.kurobbs.com/forum/getPostDetail",
        "star_post": "https://api.kurobbs.com/forum/like",
        "share_task": "https://api.kurobbs.com/encourage/level/shareTask",
        "task_process": "https://api.kurobbs.com/encourage/level/getTaskProcess",
        "total_gold": "https://api.kurobbs.com/encourage/gold/getTotalGold",
    }

    def __init__(self, token: str):
        self.token = token
        self.headers = {
            "source": "android",
            "token": token,
            "Content-Type": "application/x-www-form-urlencoded",
            "User-Agent": "okhttp/3.10.0",
            "Origin": "https://web-static.kurobbs.com",
        }

    def get_data(self, key: str, data: dict = None):
        """统一封装post请求"""
        url = self.BASE_URLS.get(key)
        try:
            response = requests.post(url, headers=self.headers, data=data, timeout=5)
            response.raise_for_status()
            return response.json()
        except requests.RequestException as e:
            return {"error": str(e)}

    def game_roles(self):
        return self.get_data("game_roles")

    def signin(self, role):
        """执行游戏签到"""
        data = {
            'gameId': role.gameId,
            'serverId': role.serverId,
            'roleId': role.roleId,
            'userId': role.userId,
            "reqMonth": time.strftime("%m")
        }
        self.get_data("init_signin", data)  # 先初始化签到
        return self.get_data("game_signin", data)

    def sign_record(self, role):
        data = {
            'gameId': role.gameId,
            'serverId': role.serverId,
            'roleId': role.roleId,
            'userId': role.userId,
        }
        return self.get_data('sign_record', data)

    def forum_signin(self):
        return self.get_data("forum_signin")

    def widget_table(self, gameId):
        if gameId == 3:
            key = "mc_widget"
            data = {"type": "2", "sizeType": "1"}
        else:
            key = "zs_widget"
            data = {"type": "1", "sizeType": "2"}
        return self.get_data(key, data)

    def zs_month(self, roleId):
        data = {"roleId": roleId}
        return self.get_data('zs_month', data)

    def forum_list(self):
        data = {
            "forumId": "9",
            "gameId": "3",
            "pageIndex": "1",
            "pageSize": "20",
            "searchType": "3",
            "timeType": "0"
        }
        return self.get_data("post_list", data)

    def forum_detail(self, post_id: str):
        data = {
            "isOnlyPublisher": "0",
            "postId": post_id,
            "showOrderTyper": "2"
        }
        return self.get_data("post_detail", data)

    def star_post(self, user_id: str, post_id: str):
        data = {
            "forumId": 11,
            "gameId": 3,
            "likeType": 1,
            "operateType": 1,
            "postCommentId": "",
            "postCommentReplyId": "",
            "postId": post_id,
            "postType": 1,
            "toUserId": user_id
        }
        return self.get_data("star_post", data)

    def share_task(self):
        return self.get_data("share_task", {"gameId": 3})

    def task_process(self):
        return self.get_data("task_process", {"gameId": "0"})

    def total_gold(self):
        return self.get_data("total_gold")

File: cfg/KuroApi.js
import fetch from 'node-fetch'


export class KuroApi {
  constructor () {
    this.token = process.env.token
    this.kuroApiUrl = 'https://api.kurobbs.com'
  }

  async getData (key = '', data = {}) {
    let { url, body = '' } = await this.getKuroApi(key, data)
    let headers = this.headers || await this.getHeaders(url)
    try {
      let response = await fetch(url, {
        method: 'post',
        headers: headers,
        body: body
      })
      if (!response.ok) {
        return { code: response.status, msg: response.statusText }
      }
      return await response.json()
    } catch (e) {
      return { code: e.code, msg: e.msg }
    }
  }

  async getKuroApi (name, data) {
    let ApiMap = {
      // 取绑定游戏账号列表
      roleList: {
        url: `${ this.kuroApiUrl }/gamer/role/default`,
      },
      // 鸣潮、战双小组件
      widget: {
        url: `${ this.kuroApiUrl }/gamer/widget/game${ data.gameId }/getData`,
        body: `type=${ data.type }&sizeType=${ data.sizeType }`
      },
      // 取签到配置信息 V2
      initSignIn: {
        url: `${ this.kuroApiUrl }/encourage/signIn/initSignInV2`,
        body: `gameId=${ data.gameId }&serverId=${ data.serverId }&roleId=${ data.roleId }&userId=${ data.userId }`,
      },
      // 游戏签到 V2
      signin: {
        url: `${ this.kuroApiUrl }/encourage/signIn/v2`,
        body: `gameId=${ data.gameId }&serverId=${ data.serverId }&roleId=${ data.roleId }&userId=${ data.userId }&reqMonth=${ data.reqMonth }`,
      },
      // 取游戏签到记录 V2
      queryRecord: {
        url: `${ this.kuroApiUrl }/encourage/signIn/queryRecordV2`,
        body: `gameId=${ data.gameId }&serverId=${ data.serverId }&roleId=${ data.roleId }&userId=${ data.userId }`,
      },
      // 战双资源
      month: {
        url: `${ this.kuroApiUrl }/haru/resource/currentMonth`,
        body: `roleId=${ data.roleId }`
      },
      // 取个人信息
      mine: {
        url: `${ this.kuroApiUrl }/user/mineV2`,
      },
      // 社区签到
      forumSignIn: {
        url: `${ this.kuroApiUrl }/user/signIn`,
        body: `gameId=${ data.gameId }`,
      },
      //取帖子列表
      forumList: {
        url: `${ this.kuroApiUrl }/forum/list`,
        body: `forumId=${ data.forumId }&gameId=${ data.gameId }&pageIndex=${ data.pageIndex }&pageSize=${ data.pageSize }`,
      },
      //取帖子详情
      postDetail: {
        url: `${ this.kuroApiUrl }/forum/getPostDetail`,
        body: `postId=${ data.postId }`,
      },
      //通用论坛点赞
      like: {
        url: `${ this.kuroApiUrl }/forum/like`,
        body: `forumId=${ data.forumId }&gameId=${ data.gameId }&postId=${ data.postId }&toUserId=${ data.toUserId }`,
      },
      // 社区分享任务
      shareTask: {
        url: `${ this.kuroApiUrl }/encourage/level/shareTask`,
        body: `gameId=${ data.gameId }&postId=${ data.postId }`,
      },
      // 取任务列表
      taskList: {
        url: `${ this.kuroApiUrl }/encourage/level/getTaskProcess`,
        body: `gameId=0&userId=${ data.userId }`,
      },
      // 取库洛币总数
      totalGold: {
        url: `${ this.kuroApiUrl }/encourage/gold/getTotalGold`,
      },
    }
    return ApiMap[name]
  }

  async getHeaders (url = '') {
    let headers = {
      Host: 'api.kurobbs.com',
      Connection: 'keep-alive',
      source: 'android',
      token: this.token,
      devCode: '171.37.188.39, Mozilla/5.0 (... Chrome/97 Mobile Safari/537.36 Kuro/2.4.2 KuroGameBox/2.4.2)',
      'Content-Type': 'application/x-www-form-urlencoded',
      'Accept-Encoding': 'gzip, deflate',
      'User-Agent': 'Mozilla/5.0 (... Chrome/97 Mobile Safari/537.36 Kuro/2.4.2 KuroGameBox/2.4.2)',
      Accept: 'application/json, text/plain, */*',
      Origin: 'https://web-static.kurobbs.com',
      'X-Requested-With': 'com.kurogame.kjq',
      'Accept-Language': 'zh-CN,zhq=0.9,en-USq=0.8,enq=0.7'
    }
    if (/(default|user|forum|level|gold)/.test(url)) {
      headers.devCode = ''
      headers.version = '2.4.2'
      headers.versionCode = '2420'
      headers.model = 'PCLM10'
      headers.Cookie = `user_token=${ this.token }`
      headers['User-Agent'] = 'okhttp/3.11.0'
    }
    return this.headers = headers
  }

}


export default new KuroApi()

